py-polars/polars/dataframe/frame.py

```diff
@@ -5743,7 +5746,7 @@ def pivot(
         values: Sequence[str] | str,
         index: Sequence[str] | str,
         columns: Sequence[str] | str,
-        aggregate_function: PivotAgg | Expr = "first",
+        aggregate_function: PivotAgg | Expr | None | NoDefault = no_default,
```

Comment:
Could you explain to me how the `NoDefault` type works?

Reply:
It's just a trick to manage the deprecation period whilst the default value changes. The idea is that by using a default value which the user wouldn't pass directly, they are "forced" to decide whether to opt-in to the new behaviour or to keep the current one

Setting `None` as the default wouldn't work because that will be the new default, so there'd be no way of telling whether `aggregate_function=None` happened because the user intentionally passed `None` or because they just used the default

Reply:
Ah.. right. That's clever. 👌